src/components/FinalizePayment/FinalizePayment.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocation, useNavigate } from 'react-router';
+
+import styles from './FinalizePayment.module.scss';
+
+import Button from '#components/Button/Button';
+import Spinner from '#components/Spinner/Spinner';
+import useEventCallback from '#src/hooks/useEventCallback';
+import useQueryParam from '#src/hooks/useQueryParam';
+import { reloadActiveSubscription } from '#src/stores/AccountController';
+import { useConfigStore } from '#src/stores/ConfigStore';
+import { replaceQueryParam, removeQueryParam, addQueryParam } from '#src/utils/location';
+import { finalizeAdyenPayment } from '#src/stores/CheckoutController';
+
+const FinalizePayment = () => {
+  const { t } = useTranslation('account');
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { accessModel } = useConfigStore(({ accessModel }) => ({ accessModel }));
+  const redirectResult = useQueryParam('redirectResult');
```

Comment:
If we can, I would prefer to use the useSearchParams hook for query parameter reading and writing.

Reply:
@ChristiaanScheermeijer I think there was a reason we are not using the useSearchParams hook van you remember why?

Reply:
I am trying to remember why... There was a reason, probably in the range of the component becoming reactive to all search parameters instead of limiting this to the search parameters the component cares about.

It is probably no issue in this component, but we need to try first.